### PR TITLE
Allow underscores in big"[float]"

### LINKED ISSUE #40368: Underscores in big_str macro for floats
```
julia> 0.123_456
0.123456

julia> big"0.123_456"
ERROR: ArgumentError: invalid number format 0.123_456 for BigInt or BigFloat
Stacktrace:
 [1] top-level scope
   @ REPL[2]:1
```

It was mentioned in https://github.com/JuliaLang/julia/pull/24834#issuecomment-348708899, but I didn't find a dedicated issue. There used to be issue (https://github.com/JuliaLang/julia/issues/21092), but that was closed when the integer case was implemented.

### PR DESCRIPTION
**Pull request by ettersi**
Fixes https://github.com/JuliaLang/julia/issues/40368

### CODE REVIEW

**Comment by ettersi** [comment]
I believe the failing test is unrelated to this PR and related to https://github.com/JuliaLang/julia/pull/43125 instead.

File: test/int.jl [hunk 1]
```diff
@@ -324,10 +324,16 @@ end
     end
 end
 
-@testset "issue #21092" begin
+@testset "Underscores in big_str" begin
     @test big"1_0_0_0" == BigInt(1000)
     @test_throws ArgumentError big"1_0_0_0_"
     @test_throws ArgumentError big"_1_0_0_0"
+
+    @test big"1_0.2_5" == BigFloat(10.25)
```
Comment (oscardssmith) [inline test/int.jl:332]: Do these tests belong in `int.jl`? I guess it doesn't really matter...

Comment (ettersi) [reply test/int.jl:332]: Fair point. It's a consequence of the `big"[float]"` code being in `base/int.jl`, which was true already before this PR.

Comment (oscardssmith) [reply test/int.jl:332]: fair enough. Not worth fixing.

**Review by oscardssmith — APPROVED** [review]

**Pull request merged**